src/components/table/index.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import TrashIcon from 'assets/icons/iconTrash';
+import ViewIcon from 'assets/icons/iconView';
+import Pagination from 'components/pagination';
+import SearchInput from 'components/searchInput';
+import TableSort from 'components/tableSort';
+import { paths } from 'const/paths';
+import { formatDate } from 'helpers/dateHelper';
+import {
+  isErrorWithMessage,
+  isFetchBaseQueryError,
+} from 'helpers/errorHandler';
+import {
+  useDeleteUserMutation,
+  useGetUsersQuery,
+} from 'redux/superAdminApiSlice';
+
+import {
+  ActionBtn,
+  Inner,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  TdInner,
+  Th,
+  Thead,
+} from './styles';
+
+// TODO add delete modal
+
+export const TableComponent: React.FC = () => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const [error, setError] = useState<string>();
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [totalPages, setTotalPages] = useState<number | null>(null);
+  const [limit] = useState<number>(5);
+  const [search, setSearch] = useState<string>('');
+  const [debouncedSearch, setDebouncedSearch] = useState(search);
+  const [order, setOrder] = useState<'ASC' | 'DESC'>('ASC');
+  const [deleteUser] = useDeleteUserMutation();
+  const { data, isLoading } = useGetUsersQuery({
+    page: currentPage,
+    limit,
+    search: debouncedSearch,
+    order,
+  });
+
+  useEffect(() => {
+    const handleSetPages = () => {
+      if (data) {
+        const totalPageCount = Math.ceil(data.totalCount / limit);
+        setTotalPages(totalPageCount);
+      }
+    };
+    handleSetPages();
+  }, [data, limit]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDebouncedSearch(search);
+    }, 1000);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [search]);
+
+  const handlePageChange = (newPage: number) => {
+    if (newPage < 1 || newPage > totalPages!) return;
+    setCurrentPage(newPage);
+  };
+
+  const handleSortChange = () => {
+    setOrder(order === 'ASC' ? 'DESC' : 'ASC');
```

Comment:
we should not compare with strings, only with constants. For these maybe we can make an enum